### DISCUSSION TITLE
Feature request - add All Ears and no Stone Unturned tracker overlay

### DISCUSSION THREAD

**Discussion by GrimDawn-max** [Lailloken/Exile-UI#192] [Ideas & Feedback]
Awesome updates as usual ! I have an idea for something that is a little off the beaten track but related to 2 achievements in the game.  There used to be a tracker for the All Ears and No Stone Unturned achievements. It looks a lot like an AHK overlay (though its c++ I think). It helped completing these achievements by tracking what a player had done/not done. It was last updated a year ago (the code only since the achievements have not changed since 3.17). Sadly it no longer works. I reached out to the author but had no response. Would it be possible to incorporate this into your tool ?

The github repository in question can be found here - https://github.com/Nickswoboda/all-ears-unturned
Here's hoping ...

**Comment by Lailloken** [comment]
Thanks for your suggestion. I just had a look at the tool and I think there's no way to implement something similar without straight up copying/stealing what they did, which I'm not really interested in. Whenever I implement something, I do it either because conceptualizing or working on it is interesting/fun in itself or because I have additional ideas I want to incorporate -- neither of which is the case here.

Most of the work is collecting the information and putting together the guide, which has already been done and which I wouldn't want to take advantage of.

**Reply by GrimDawn-max** [reply]
That's fair. I just found a browser based version - https://github.com/halfacandan/PoEUserScripts 
Should be able to use that. Any plans or timeline on updating the lab section so this could be used to find your way around, at least in Uber. I know there is a compass like app called labbie but I find I spend too much time looking at the compass and not where I'm going in that :-)

**Reply by Lailloken** [reply]
I changed my mind about the lab overlay. I'll improve usability as well as importing the current layout, but I won't make it more than a "mini-map" for casual lab-runs. Anything more would make comparisons with the lab compass inevitable -- and a sub-feature of mine would always pale in comparison with a single-focus tool (again, I also don't want to just copy what it does).

I think I found a way to keep the regular poelab.com image (I personally think it's sufficient for my lab needs) and automatically highlight the room you're currently in and also overlay the names of the neighboring rooms. This way, the user knows the rough directions to the next room and the name (so they won't have to check the in-game room-plan in front of each exit).

I haven't put much thought into it other than that.

**Comment by Lailloken** [comment]
Needs a bit more testing, but here's my progress:

![image](https://user-images.githubusercontent.com/61888437/219698569-fcb51a8e-66f0-4845-aa8a-89b645929a8d.png)

The green markers are the rooms you've been through, red the current one. It shows abbreviations of the neighboring rooms' names. I also changed overlay-activation to holding TAB, and hiding it on release.

If you have any more ideas, let me know.

**Comment by GrimDawn-max** [comment]
This looks great. At least now I can remember where I’ve been and where I am 👍 which is a great visual and should prevent back tracking unnecessarily.

I have one other though, not lab related. Is it possible to track when you kill a map boss ? Reason I ask is in open areas I often forget I’ve killed the boss. Sometimes they are killed off screen. If they drop nothing and don’t give any map craft rewards there’s no clue. If there was a “boss killed” icon in maps or something to indicate you’ve done it that would be a nice addition. I’m not sure if it’s in the Client.txt file. Didn’t spot anything but could have missed it to.

**Reply by Lailloken** [reply]
I don't think there's a way to track that.
